### ISSUE TITLE
Duckdb double equals: unparsable section: "CASE | WHEN

### ISSUE THREAD

**Issue by ChristianRothQC** [sqlfluff/sqlfluff#6905]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

```sql
CREATE TABLE my_table (
    col1 TEXT,
    cola INTEGER,
    colb INTEGER
);

INSERT INTO my_table (col1, cola, colb) VALUES
('A', 10, NULL),
('B', NULL, 20);

SELECT
    COALESCE(
        MAX(CASE WHEN col1 == 'A' THEN cola END),
        MAX(CASE WHEN col1 == 'B' THEN colb END),
        MAX(CASE WHEN col1 == 'C' THEN colb END)
    ) AS result
FROM my_table
```

Seems valid duckdb sql (note the double equals sign in CASE WHEN), but sqlfluff does not seem to like it. Changing the double equals to  single equals makes sqlfluff happy again:

```sql
CREATE TABLE my_table (
    col1 TEXT,
    cola INTEGER,
    colb INTEGER
);

INSERT INTO my_table (col1, cola, colb) VALUES
('A', 10, NULL),
('B', NULL, 20);

SELECT
    COALESCE(
        MAX(CASE WHEN col1 = 'A' THEN cola END),
        MAX(CASE WHEN col1 = 'B' THEN colb END),
        MAX(CASE WHEN col1 = 'C' THEN colb END)
    ) AS result
FROM my_table
```

### Expected Behaviour

Have a happy linter despite double == comparisons.

### Observed Behaviour

```
L:  13 | P:   9 | CP02 | Unquoted identifiers must be consistently lower case.
                       | [capitalisation.identifiers]
L:  13 | P:  12 | LT01 | Expected single whitespace between naked identifier and
                       | start bracket '('. [layout.spacing]
L:  13 | P:  13 |  PRS | Line 13, Position 13: Found unparsable section: "CASE
                       | WHEN col1 == 'A' THEN cola END"
L:  14 | P:   9 | CP02 | Unquoted identifiers must be consistently lower case.
                       | [capitalisation.identifiers]
L:  14 | P:  12 | LT01 | Expected single whitespace between naked identifier and
                       | start bracket '('. [layout.spacing]
L:  14 | P:  13 |  PRS | Line 14, Position 13: Found unparsable section: "CASE
                       | WHEN col1 == 'B' THEN colb END"
L:  15 | P:   9 | CP02 | Unquoted identifiers must be consistently lower case.
                       | [capitalisation.identifiers]
L:  15 | P:  12 | LT01 | Expected single whitespace between naked identifier and
                       | start bracket '('. [layout.spacing]
L:  15 | P:  13 |  PRS | Line 15, Position 13: Found unparsable section: "CASE
                       | WHEN col1 == 'C' THEN colb END"
WARNING: Parsing errors found and dialect is set to 'duckdb'. Have you configured your dialect correctly?
```

### How to reproduce

`sqlfluff lint test.sql --dialect duckdb`

### Dialect

duckdb

### Version

sqlfluff, version 3.4.0

### Configuration

none

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
